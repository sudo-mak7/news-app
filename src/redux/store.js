import { legacy_createStore as createStore, compose, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import rootReducer from './reducers/rootReducer'
import { newsWatcher } from './sagas/rootSaga'

const sagaMiddleware = createSagaMiddleware()

const composeEnhancers =
  typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose

const configureStore = (preloadedState) => createStore(
  rootReducer,
  preloadedState,
  composeEnhancers(applyMiddleware(sagaMiddleware))
)

const store = configureStore({})

sagaMiddleware.run(newsWatcher)

export default store
