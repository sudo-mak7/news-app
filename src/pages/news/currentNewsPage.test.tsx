import * as React from 'react'
import { screen, render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { store } from '@redux/store'
import { Provider } from 'react-redux'
import CurrentNewsPage from '@pages/news/currentNewsPage'

const currentNewsPage = (): JSX.Element =>
  <Provider store = { store } >
    <BrowserRouter>
      <CurrentNewsPage />
    </BrowserRouter>
  </Provider>

describe('CurrentNewsPage test', () => {
  test('CurrentNewsPage should to match snapshot', () => {
    window.scrollTo = jest.fn()

    render(currentNewsPage())
    expect(currentNewsPage()).toMatchSnapshot()
  })

  test('renders CurrentNewsPage ButtonBackToNewsComponent', () => {
    window.scrollTo = jest.fn()

    render(currentNewsPage())

    const buttonInCurrentNewsPage = screen.getByRole('button')
    expect(buttonInCurrentNewsPage).toBeInTheDocument()
    expect(buttonInCurrentNewsPage).toHaveClass('ui blue big icon button')
  })

  test('renders CurrentNewsPage news block', () => {
    window.scrollTo = jest.fn()

    render(currentNewsPage())

    const articleInCurrentNewsPage = screen.getByRole('article')
    expect(articleInCurrentNewsPage).toBeInTheDocument()
  })

  afterAll(() => {
    jest.clearAllMocks()
  })
})
