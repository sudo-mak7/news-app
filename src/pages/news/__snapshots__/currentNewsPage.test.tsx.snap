// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentNewsPage test CurrentNewsPage should to match snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <CurrentNewsPage />
  </BrowserRouter>
</Provider>
`;

exports[`CurrentNewsPage test renders CurrentNewsPage ButtonBackToNewsComponent 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <CurrentNewsPage />
  </BrowserRouter>
</Provider>
`;

exports[`CurrentNewsPage test renders CurrentNewsPage news block 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <CurrentNewsPage />
  </BrowserRouter>
</Provider>
`;
